src/api/GameContext.tsx

```diff
@@ -26,24 +39,68 @@ type DeleteGameAction = {
   type: 'DELETE_GAME'
 }
 
-type Action = SetCardsAction | DeleteGameAction
+type FinishGameAction = {
+  type: 'FINISH_GAME'
+}
+
+type UpdateButtonAction = {
+  type: 'UPDATE_BUTTON'
+}
+
+type Action =
+  | StartGameAction
+  | SetCardIndexAction
+  | SetCardsAction
+  | DeleteGameAction
+  | FinishGameAction
+  | UpdateButtonAction
 
 type Props = {
   state: State
   dispatch: React.Dispatch<Action>
+  handleButtonClick: () => void
 }
 
 const initialState: State = {
   cards: [],
-  isGameOngoing: false,
+  gameState: GameState.NOT_STARTED,
+  buttonLabel: 'New Game',
+  currentCardIndex: 0,
 }
 
 const gameReducer = (state: State, action: Action): State => {
   switch (action.type) {
+    case 'START_GAME':
+      return { ...state, gameState: GameState.START, currentCardIndex: 0 }
+    case 'SET_CARD_INDEX':
+      return { ...state, currentCardIndex: action.payload }
     case 'SET_CARDS':
-      return { ...state, cards: action.payload, isGameOngoing: true }
+      return {
+        ...state,
+        cards: action.payload,
+        gameState: GameState.ONGOING,
+        currentCardIndex: 0,
+      }
```

Comment:
Auch hier etwas zweispurig, weil START_GAME in etwa das geliche macht und dann auf ONGOING resultieren soll

Reply:
done in #41